include/circt/Dialect/Ibis/IbisOps.td

```diff
@@ -160,13 +167,48 @@ def CallOp : IbisOp<"call", [CallOpInterface]> {
   }];
 }
 
+def GetParentOfRefOp : IbisOp<"get_parent_of_ref"> {
+  let summary = "Ibis get classref parent op";
+  let description = [{
+    Given an Ibis class reference, returns a class reference to the parent of
+    said class. The returned `ibis.classref` will be an opaque type.
+  }];
+
+  let arguments = (ins AnyClassRefType:$src);
+  let results = (outs OpaqueClassRefType:$parent);
+
+  let assemblyFormat = [{
+    $src `:` qualified(type($src)) attr-dict
+  }];
+}
+
+def GetInstanceInRefOp : IbisOp<"get_instance_in_ref"> {
+  let summary = "Ibis get instance";
+  let description = [{
+    Given an Ibis class reference, returns a instance of said class. The instance
+    is specified by the symbol name of the instance of that instance in the referenced
```

Comment:
Typo, should it be "symbol name of the instance in the instance in the referenced class.